src/serialize.cpp

```diff
@@ -116,13 +133,16 @@ namespace sparrow_ipc
             batch_builder.Finish(batch_message_offset);
 
             // Append the RecordBatch message to the final buffer
-            const uint32_t batch_meta_len = batch_builder.GetSize(); // Get the size of the batch metadata
-            const int64_t aligned_batch_meta_len = utils::align_to_8(batch_meta_len); // Calculate the padded length
+            const uint32_t batch_meta_len = batch_builder.GetSize();  // Get the size of the batch metadata
+            const int64_t aligned_batch_meta_len = utils::align_to_8(batch_meta_len);  // Calculate the padded
+                                                                                       // length
 
-            const size_t current_size = final_buffer.size(); // Get the current size (which is the end of the Schema message)
+            const size_t current_size = final_buffer.size();  // Get the current size (which is the end of the
+                                                              // Schema message)
             // Resize the buffer to append the new message
             final_buffer.resize(current_size + sizeof(uint32_t) + aligned_batch_meta_len + body_len);
-            uint8_t* dst = final_buffer.data() + current_size; // Get a pointer to where the new message will start
+            uint8_t* dst = final_buffer.data() + current_size;  // Get a pointer to where the new message will
+                                                                // start
 
             // Write the 4-byte metadata length for the RecordBatch message
             *(reinterpret_cast<uint32_t*>(dst)) = batch_meta_len;
```

Comment:
About the code below: the null bitmap is optional in the IPC stream as well, so the `memset` thing is really sub-optimal :)

Reply:
removed